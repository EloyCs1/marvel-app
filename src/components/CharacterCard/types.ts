import { AppCharacter } from "src/service/types";

export interface CharacterCardProps {
  character: AppCharacter.Character;
}
