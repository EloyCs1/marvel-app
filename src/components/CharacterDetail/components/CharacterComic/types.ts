import { AppCharacter } from "src/service/types";

export interface CharacterComicProps {
  comic: AppCharacter.Comic;
}
