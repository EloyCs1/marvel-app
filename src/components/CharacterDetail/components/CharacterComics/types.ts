import { AppCharacter } from "src/service/types";

export interface CharacterComicsProps {
  comics: AppCharacter.Comic[];
  loading: boolean;
}
