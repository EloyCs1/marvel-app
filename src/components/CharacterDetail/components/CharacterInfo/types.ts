import { AppCharacter } from "src/service/types";

export interface CharacterInfoProps {
  detail: AppCharacter.Character;
  loading: boolean;
}
