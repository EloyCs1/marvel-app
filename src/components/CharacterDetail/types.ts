export interface CharacterDetailProps {
  characterId: number;
}
