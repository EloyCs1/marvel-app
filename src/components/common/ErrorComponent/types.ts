export interface ErrorComponentProps {
  testId?: string;
  tryAgain?: () => void;
}
