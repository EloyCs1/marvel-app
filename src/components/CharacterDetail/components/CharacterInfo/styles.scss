.characterDetail {
  &__info {
    background: var(--color-black);
    color: var(--color-white);
    display: flex;
    justify-content: center;
    gap: 48px;
    position: relative;

    img {
      max-width: 320px;
    }

    &-content {
      width: 592px;
      display: grid;
      align-content: center;
      gap: 24px;

      &-header {
        display: flex;
        justify-content: space-between;
        height: 38px;
        align-items: center;

        p {
          font-size: 32px;
          line-height: 37.5px;
          width: fit-content;
        }
      }
    }

    &:before {
      content: "";
      position: absolute;
      bottom: 0;
      right: 0;
      border-bottom: 24px solid var(--color-white);
      border-left: 24px solid var(--color-black);
      width: 0;
    }

    @media (max-width: 960px) {
      img {
        max-width: 278px;
      }
      &-content {
        padding-right: 48px;
      }
    }

    @media (max-width: 575px) {
      display: grid;

      img {
        max-width: 100%;
      }

      &-content {
        width: calc(100% - 32px);
        padding-right: 16px;
        padding-left: 16px;
        padding-bottom: 48px;

        &-header {
          height: unset;
        }
      }
    }
  }
}
