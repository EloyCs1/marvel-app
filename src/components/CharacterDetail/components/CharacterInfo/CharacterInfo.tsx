import { IconName } from "src/components/common/Icon/types";
import Icon from "src/components/common/Icon/Icon";
import Loader from "src/components/common/Loader/Loader";
import { useFavorites } from "src/context/FavoriteContext";

import { CharacterInfoProps } from "./types";
import "./styles.scss";

const CharacterInfo: React.FC<CharacterInfoProps> = ({ detail, loading }) => {
  const { id, name, description, image } = detail;
  const { favoriteIds, toggleFavoriteId } = useFavorites();

  const nameIcon = favoriteIds.includes(id) ? IconName.FAVORITES_FILLED : IconName.FAVORITES_OUTLINED;
  const onClickIcon = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    e.preventDefault();
    e.stopPropagation();
    toggleFavoriteId(id);
  };

  return (
    <div className="characterDetail__info">
      <div className="characterDetail__info-image" style={{ backgroundImage: `url(${image})` }}>
        {loading && <Loader />}
      </div>
      <div className="characterDetail__info-content">
        <div className="characterDetail__info-content-header">
          <p>{name}</p>
          {!loading && <Icon name={nameIcon} onClick={onClickIcon} />}
        </div>
        <p>{description}</p>
      </div>
    </div>
  );
};
export default CharacterInfo;
