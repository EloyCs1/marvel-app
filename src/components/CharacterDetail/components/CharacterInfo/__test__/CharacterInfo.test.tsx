import { render } from "@testing-library/react";
import { CHARACTER } from "src/mocks/mocks";
import { AppWrapper } from "src/test/test.utils";
import { CharacterInfoProps } from "../types";
import CharacterInfo from "../CharacterInfo";

const defaultProps: CharacterInfoProps = {
  detail: CHARACTER,
  loading: false,
};

describe("CharacterInfo", () => {
  test("Default render", () => {
    const wrapper = render(
      <AppWrapper>
        <CharacterInfo {...defaultProps} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("Loading render", () => {
    const wrapper = render(
      <AppWrapper>
        <CharacterInfo {...defaultProps} loading={true} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
