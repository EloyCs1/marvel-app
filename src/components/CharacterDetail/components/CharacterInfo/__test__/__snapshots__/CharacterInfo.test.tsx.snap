// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterInfo Default render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="characterDetail__info"
      >
        <div
          class="characterDetail__info-image"
          style="background-image: url(http://i.annihil.us/u/prod/marvel/i/mg/3/20/5232158de5b16.jpg);"
        />
        <div
          class="characterDetail__info-content"
        >
          <div
            class="characterDetail__info-content-header"
          >
            <p>
              A-Bomb (HAS)
            </p>
            <button
              class="buttonIcon"
            >
              <svg
                fill="none"
                height="21.68"
                viewBox="0 0 15 14"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.75 1.55237L5.26885 0.697502L4.75 0.382594L4.23115 0.697502L4.75 1.55237ZM7.75 3.37318L7.23115 4.22804L7.75 4.54295L8.26885 4.22804L7.75 3.37318ZM1.75 3.37318L1.23115 2.51831L0.75 2.81034V3.37318H1.75ZM1.75 7.27491H0.75V7.73644L1.1012 8.03587L1.75 7.27491ZM7.75 12.3905L7.1012 13.1515L7.75 13.7046L8.3988 13.1515L7.75 12.3905ZM13.75 7.27491L14.3988 8.03587L14.75 7.73644V7.27491H13.75ZM13.75 3.37318H14.75V2.81034L14.2688 2.51831L13.75 3.37318ZM10.75 1.55237L11.2689 0.697502L10.75 0.382594L10.2312 0.697502L10.75 1.55237ZM4.23115 2.40723L7.23115 4.22804L8.26885 2.51831L5.26885 0.697502L4.23115 2.40723ZM2.26885 4.22804L5.26885 2.40723L4.23115 0.697502L1.23115 2.51831L2.26885 4.22804ZM2.75 7.27491V3.37318H0.75V7.27491H2.75ZM8.3988 11.6296L2.3988 6.51395L1.1012 8.03587L7.1012 13.1515L8.3988 11.6296ZM8.3988 13.1515L14.3988 8.03587L13.1012 6.51395L7.1012 11.6296L8.3988 13.1515ZM14.75 7.27491V3.37318H12.75V7.27491H14.75ZM14.2688 2.51831L11.2689 0.697502L10.2312 2.40723L13.2312 4.22804L14.2688 2.51831ZM10.2312 0.697502L7.23115 2.51831L8.26885 4.22804L11.2689 2.40723L10.2312 0.697502Z"
                  fill="white"
                />
              </svg>
            </button>
          </div>
          <p>
            Rick Jones has been Hulk's best bud since day one 
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="characterDetail__info"
    >
      <div
        class="characterDetail__info-image"
        style="background-image: url(http://i.annihil.us/u/prod/marvel/i/mg/3/20/5232158de5b16.jpg);"
      />
      <div
        class="characterDetail__info-content"
      >
        <div
          class="characterDetail__info-content-header"
        >
          <p>
            A-Bomb (HAS)
          </p>
          <button
            class="buttonIcon"
          >
            <svg
              fill="none"
              height="21.68"
              viewBox="0 0 15 14"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.75 1.55237L5.26885 0.697502L4.75 0.382594L4.23115 0.697502L4.75 1.55237ZM7.75 3.37318L7.23115 4.22804L7.75 4.54295L8.26885 4.22804L7.75 3.37318ZM1.75 3.37318L1.23115 2.51831L0.75 2.81034V3.37318H1.75ZM1.75 7.27491H0.75V7.73644L1.1012 8.03587L1.75 7.27491ZM7.75 12.3905L7.1012 13.1515L7.75 13.7046L8.3988 13.1515L7.75 12.3905ZM13.75 7.27491L14.3988 8.03587L14.75 7.73644V7.27491H13.75ZM13.75 3.37318H14.75V2.81034L14.2688 2.51831L13.75 3.37318ZM10.75 1.55237L11.2689 0.697502L10.75 0.382594L10.2312 0.697502L10.75 1.55237ZM4.23115 2.40723L7.23115 4.22804L8.26885 2.51831L5.26885 0.697502L4.23115 2.40723ZM2.26885 4.22804L5.26885 2.40723L4.23115 0.697502L1.23115 2.51831L2.26885 4.22804ZM2.75 7.27491V3.37318H0.75V7.27491H2.75ZM8.3988 11.6296L2.3988 6.51395L1.1012 8.03587L7.1012 13.1515L8.3988 11.6296ZM8.3988 13.1515L14.3988 8.03587L13.1012 6.51395L7.1012 11.6296L8.3988 13.1515ZM14.75 7.27491V3.37318H12.75V7.27491H14.75ZM14.2688 2.51831L11.2689 0.697502L10.2312 2.40723L13.2312 4.22804L14.2688 2.51831ZM10.2312 0.697502L7.23115 2.51831L8.26885 4.22804L11.2689 2.40723L10.2312 0.697502Z"
                fill="white"
              />
            </svg>
          </button>
        </div>
        <p>
          Rick Jones has been Hulk's best bud since day one 
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CharacterInfo Loading render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="characterDetail__info"
      >
        <div
          class="characterDetail__info-image"
          style="background-image: url(http://i.annihil.us/u/prod/marvel/i/mg/3/20/5232158de5b16.jpg);"
        >
          <div
            class="loader"
          >
            <span
              class="loader__spinner"
            />
          </div>
        </div>
        <div
          class="characterDetail__info-content"
        >
          <div
            class="characterDetail__info-content-header"
          >
            <p>
              A-Bomb (HAS)
            </p>
          </div>
          <p>
            Rick Jones has been Hulk's best bud since day one 
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="characterDetail__info"
    >
      <div
        class="characterDetail__info-image"
        style="background-image: url(http://i.annihil.us/u/prod/marvel/i/mg/3/20/5232158de5b16.jpg);"
      >
        <div
          class="loader"
        >
          <span
            class="loader__spinner"
          />
        </div>
      </div>
      <div
        class="characterDetail__info-content"
      >
        <div
          class="characterDetail__info-content-header"
        >
          <p>
            A-Bomb (HAS)
          </p>
        </div>
        <p>
          Rick Jones has been Hulk's best bud since day one 
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
